### PR TITLE
Enclave rework v5

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About The Pull Request

- Gives Enclave a new bunker in a new location. CAR-15s have been removed and replaced with 10mm SMGs, 2 AER12s and one AER14.
- Adds an "indoctrinator" device, that turns people into Enclave Recruits if they consent to it. Scientists can no longer brainwash.
- Adds a small ruin in the place of old Enclave base with low tier loot.
- Removes brainwashing abilities from Enclave scientist.

## Why It's Good For The Game

Most of the time I asked people "What is wrong with the Enclave?" I heard:
1. CAR-15 too OP, remove pls.
2. Brainwash is a bad/ungodly/terrible/boring/sad mechanic and should be gone.

So here it is. Changing these things as much as I could.

enclavebasefull
enclavebase

Other than obvious, mechanic changes, most people wanted Enclave to stay, so 60% of the players would be surely glad to see it back, albeit changed a bit.

## Tokoko's checklist
Here is the list of Enclave issues you could've seen in #player-voting and my opinion on that:

> The Enclave have Three sets of APA

They have 2, and they can no longer dupe it.

> The ability to brainwash and created more members to hypothetically have infinite members

No longer an infinite number of recruits.

> A functioning full RnD system

I don't know how this is an issue, personally, it only gives outlaws an opportunity to get technology without raiding BoS. Heck, if there were some roleplaying outlaws, they could trade with the Enclave.

>A functioning chem setup

Not an issue when half the server has it.

> Botany
Oh no, they can grow pickles, we're doomed. I really don't know how's that an issue. If anything, we should just nerf botany and remove dangerous shit from it.

> Start with some of the best rifles in the game, as well as 50 round magazines

No longer they spawn with it.

> They contentiously ally with the Legion, even if it is not lore friendly

Admin issue. They are not supposed nor allowed to ally with any other major faction.

> They raid the Brotherhood for the only reason of "Were enemies in lore" despite them saying they are supposed to be covert in their operations

So does every other faction to each other. Have you ever seen Legion wait for an IC reason to raid NCR? What about BoS raiding Enclave? Admin issue.

> They have meta protection from being raided

Yes, otherwise they'd be no-escalation raided every round when it hits 1:30.

> They have been seen dupping APA multiple times

Again, no longer possible.

> They have consistently broken rules each time I have been online moderating and only join conflicts not for the sake of RP, but for a valid count.

Seriously, admin issue. If you can't look after **4 players**, you got some serious troubles. I did my part by increasing time requirements and adding role expectations where those have been missing. Everything else is your "job".
If an Enclave Sergeant acts like a complete LRP shitter you should treat him the same as if it was an outlaw acting like complete LRP shitter.

## Changelog
:cl:
add: New Enclave bunker.
add: Indoctrinator device.
del: Enclave Scientists can no longer brainwash people.
/:cl:

### CODE REVIEW

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [inline code/game/objects/items/indoctrinator.dm:27]: Make it check for user intent, and to call parent if on harm intent.

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [inline code/game/objects/items/indoctrinator.dm:34]: You already assumed `M` exists, so no need to check twice. It would runtime if it didn't.

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [inline code/game/objects/items/indoctrinator.dm:43]: If they have no key, they have no client. No need to check for both.

Comment (EgorDinamit) [reply code/game/objects/items/indoctrinator.dm:43]: The examine.dm of humans uses it, to decide if human is catatonic or simply SSD, hence why I used it like that here.

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [inline code/game/objects/items/indoctrinator.dm:51]: From the moment the timer begins and onwards you want to stop the attack chain, as you've already started an actual action.
You can put a `. = STOP_ATTACK_PROC_CHAIN` at this point for that to happen (before the do after).

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [inline code/game/objects/items/indoctrinator.dm:54]: These two should be visible messages, shouldn't they?

Comment (EgorDinamit) [reply code/game/objects/items/indoctrinator.dm:54]: It is supposed to be visible only to the user, not everyone around.

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [reply code/game/objects/items/indoctrinator.dm:43]: Right, but if they want to know whether it's a player mob, disconnected or not, they can examine.
In here you just need to filter out the invalid candidates.

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [reply code/game/objects/items/indoctrinator.dm:54]: Alright.

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [reply code/game/objects/items/indoctrinator.dm:51]: Also make use of `do_afters` or `targeted_by` to avoid multiple actions.

**Comment by Mannybrado** [comment]
this still doesn't fix the LRP nature of the enclave, which is the main reason why they were removed. they should not have this gear, they shouldn't even have a bunker. they should be _remnants_, if they even exist at all anyways.

**Comment by Burtuu** [comment]
code it where they spawn in the middle of bumfuck nowhere and have to contact each other to get the the bunker

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [inline code/game/objects/items/indoctrinator.dm:28]:
```suggestion
		return ..()
```
Always preserve parent call return unless you have a reason not to.

File: code/game/objects/items/indoctrinator.dm
Comment (Rohesie) [inline code/game/objects/items/indoctrinator.dm:53]: Re-thinking it, this should go before any `to_chat()` message happens, since those are events already.
Just put this before `if(in_use)`

**Comment by AdamElTablawy** [comment]
This doesn't really fix any of the problems with the faction. Especially being that it really only exacerbates the problem with brainwashing-that it's stale and boring and only used for making more recruits rather than anything more versatile and unique.

This doesn't really fix any of the LRP problems. If all they needed was a nerf, they'd just have been nerfed.

Until they're more comprehensively reworked into something completely different than they once were, this doesn't work.

The summary of this change is that you made brainwashing more stale but opt in, and gave them weaker guns, and that's it. Nothing to do with any of the other problems.

**Comment by SpaceSlug2** [comment]
i want my sneed back

**Comment by EgorDinamit** [comment]
> This doesn't really fix any of the problems with the faction. Especially being that it really only exacerbates the problem with brainwashing-that it's stale and boring and only used for making more recruits rather than anything more versatile and unique.
> 
> This doesn't really fix any of the LRP problems. If all they needed was a nerf, they'd just have been nerfed.
> 
> Until they're more comprehensively reworked into something completely different than they once were, this doesn't work.
> 
> The summary of this change is that you made brainwashing more stale but opt in, and gave them weaker guns, and that's it. Nothing to do with any of the other problems.

I'm sorry, but your comment didn't make any sense, could you elaborate on what kind of issues Enclave had to begin with?
If your only concern that it is "boring", than you should consider other players, that do not think like you.

Recruiting people is the only thing that keeps Enclave different and unique from the other factions, and all in all it only means that they will get to their "full slots" later in the game.

**Comment by EgorDinamit** [comment]
> this still doesn't fix the LRP nature of the enclave, which is the main reason why they were removed. they should not have this gear, they shouldn't even have a bunker. they should be _remnants_, if they even exist at all anyways.

LRP nature? No, this wasn't the reason they got disabled.
I truly hate to remind you that Encave has numerous bases everywhere in the US, and in FO:NV they got much more equipment for combat.

Then again, our lore can be anything we want, we're not supposed to follow FO lore step by step.

**Comment by AdamElTablawy** [comment]
It wasn't really meant to be a comment left for you so much as it was for Baron, tbh.

**Comment by EgorDinamit** [comment]
> It wasn't really meant to be a comment left for you so much as it was for Baron, tbh.

I am the one doing this rework, so you don't need to leave a comment for someone else.
If you want Baron's attention, you could ping them on Discord.

**Review by thedarkcrusader — APPROVED** [review]

**Comment by Tokorizo** [comment]
No

**Comment by EgorDinamit** [comment]
> No

Yes. In the name of Enclave, in the name of democracy.
image

**Comment by SpaceSlug2** [comment]
> this still doesn't fix the LRP nature of the enclave, which is the main reason why they were removed. they should not have this gear, they shouldn't even have a bunker. they should be _remnants_, if they even exist at all anyways.

Words from a man who doesn't know lore.

**Comment by Singularity69** [comment]
Aren't AER's worse than most NCR loadout guns?

**Comment by EgorDinamit** [comment]
> Aren't AER's worse than most NCR loadout guns?

AERs are standard-issued guns to BoS knights/paladins, hence why I used these.
They might have less damage, but they compensate it with somewhat good AP.

Regardless, these are surely worse than CAR-15, which was one of the main reasons people shit-talked Enclave in the first place.

**Comment by Singularity69** [comment]
> > Aren't AER's worse than most NCR loadout guns?
> 
> AERs are standard-issued guns to BoS knights/paladins, hence why I used these.
> They might have less damage, but they compensate it with somewhat good AP.
> 
> Regardless, these are surely worse than CAR-15, which was one of the main reasons people shit-talked Enclave in the first place.

I was hoping sarges would get CAR-15's

**Comment by EgorDinamit** [comment]
> > > Aren't AER's worse than most NCR loadout guns?
> > 
> > 
> > AERs are standard-issued guns to BoS knights/paladins, hence why I used these.
> > They might have less damage, but they compensate it with somewhat good AP.
> > Regardless, these are surely worse than CAR-15, which was one of the main reasons people shit-talked Enclave in the first place.
> 
> I was hoping sarges would get CAR-15's

Had to go on some compromises to have Enclave back, I'd say.

**Comment by Mannybrado** [comment]
> LRP nature? No, this wasn't the reason they got disabled.
> I truly hate to remind you that Encave has numerous bases everywhere in the US, and in FO:NV they got much more equipment for combat.
> 
> Then again, our lore can be anything we want, we're not supposed to follow FO lore step by step.

yes... yes it was. the server is attempting, now, i believe, after all of this dogshit, to stay as lore friendly as possible to things that already exist.
image
These are the issues with the enclave, worded pretty well. This fixes one and a half of them.

**Comment by Mannybrado** [comment]
> > this still doesn't fix the LRP nature of the enclave, which is the main reason why they were removed. they should not have this gear, they shouldn't even have a bunker. they should be _remnants_, if they even exist at all anyways.
> 
> Words from a man who doesn't know lore.

elaborate, lmfao

**Comment by Rohesie** [comment]
Map merge conflict.

**Comment by EgorDinamit** [comment]
> Map merge conflict.

I'll fix these when I get a clear approval from maintainers on merging it, so I don't fix map conflicts 10 times a week.

**Comment by SpaceSlug2** [comment]
> > > this still doesn't fix the LRP nature of the enclave, which is the main reason why they were removed. they should not have this gear, they shouldn't even have a bunker. they should be _remnants_, if they even exist at all anyways.
> > 
> > 
> > Words from a man who doesn't know lore.
> 
> elaborate, lmfao

The Enclave have vertibird refueling stations EVERYWHERE filled with APA and plasma rifles. You’re telling me they wouldn’t have a SINGLE bunker in Yuma?

**Comment by EgorDinamit** [comment]
> > > > this still doesn't fix the LRP nature of the enclave, which is the main reason why they were removed. they should not have this gear, they shouldn't even have a bunker. they should be _remnants_, if they even exist at all anyways.
> > > 
> > > 
> > > Words from a man who doesn't know lore.
> > 
> > 
> > elaborate, lmfao
> 
> The Enclave have vertibird refueling stations EVERYWHERE filled with APA and plasma rifles. You’re telling me they wouldn’t have a SINGLE bunker in Yuma?

We all agree with that, but this PR isn't really about the lore of Fallout, it's more about the balance and how the game feels.

**Comment by EgorDinamit** [comment]
Alright, it's ready to go.

**Comment by ogrigbe** [comment]
Hold this thought, keep the PR and discuss it with myself and the staff team at a later date. Enclave was removed purposefully because of the sheer amount of idiots who were playing it, joining with major factions with advanced gear, kidnapping and ruining peoples games by brainwashing and torturing people. We aren't in a state where I believe the roleplay is to a standard yet where the enclave can make a return. Because they have R&D, because they have power armor, because they do not keep to themselves like the brotherhood does they've become a problem just like how the vault was. For now this gets closed, in the future it may return but we're not ready for something like this.

**Comment by EgorDinamit** [comment]
Today we enter the dark age...

**Pull request closed without merging**
